src/MetadataTools.jl

```diff
@@ -146,10 +176,9 @@ function get_upper_limit(pkg::PkgMeta)
         julia_max_ver = v"0.0.0"
         # Check if there is a Julia max version dependency
         for req in ver.requires
-            !contains(req,"julia") && continue
-            s = split(req," ")
-            length(s) != 3 && continue
-            julia_max_ver = convert(VersionNumber,s[3])
+            req.name!="julia" && continue
+            isnull(req.upper_bound) && continue
+            julia_max_ver = get(req.upper_bound)
```

Comment:
Looks like the Requirement type has a `versions` field that's a `VersionSet` type. `VersionSet` has an `intervals` field which is a vector of `VersionInterval`s. If the vector isn't length 1, the code in this package probably won't handle it correctly. The first element of the array will have a `lower` and `upper` field.